pkg/kn/commands/revision/delete.go

```diff
@@ -67,7 +92,21 @@ func NewRevisionDeleteCommand(p *commands.KnParams) *cobra.Command {
 			return nil
 		},
 	}
+	flags := RevisionDeleteCommand.Flags()
+	flags.Bool("prune", false, "Remove all unreferenced revisions in a namespace.")
 	commands.AddNamespaceFlags(RevisionDeleteCommand.Flags(), false)
 	waitFlags.AddConditionWaitFlags(RevisionDeleteCommand, commands.WaitDefaultTimeout, "delete", "revision", "deleted")
 	return RevisionDeleteCommand
 }
+
+func getRevisionNames(client v1.KnServingClient) ([]string, error) {
+	revisionList, err := client.ListRevisions()
+	if err != nil {
+		return []string{}, err
+	}
+	revisionNames := []string{}
+	for _, revision := range revisionList.Items {
+		revisionNames = append(revisionNames, revision.Name)
+	}
+	return revisionNames, nil
+}
```

Comment:
Where are we checking for unreferenced revisions here ?

This function needs to find out the subset of revisions present in this namespace which are not referenced by *any* service and return them.